pkg/store/store.go

```diff
@@ -411,21 +411,18 @@ func (store *OpenTSDBStore) composeOpenTSDBQuery(req *storepb.SeriesRequest) (op
 	var tagFilters []opentsdb.Filter
 	var metricNames []string
 	var err error
+	// Find __name__ first, if the names aren't whitelisted on Geras then we
```

Comment:
Can we try and phrase this without the negation? Like “We only want to return an error when we need to use Geras and the matcher is not compatible. So check __name__ to see if it is a Geras-compatible name before checking the matcher.”

Reply:
Done.